corehq/apps/app_manager/tasks.py

```diff
@@ -91,3 +93,57 @@ def update_linked_app_and_notify_task(domain, app_id, master_app_id, user_id, em
 def load_appcues_template_app(domain, username, app_slug):
     from corehq.apps.app_manager.views.apps import load_app_from_slug
     load_app_from_slug(domain, username, app_slug)
+
+
+@task(queue='background_queue', ignore_result=True)
+def analyse_new_app_build(domain, app_id):
+    new_build_id = get_latest_build_id(domain, app_id)
+    new_build = get_app(domain, new_build_id)
+
+    check_for_custom_callouts(new_build)
+    check_build_dependencies(new_build)
+
+
+def check_for_custom_callouts(new_build):
+    from corehq.apps.app_manager.util import app_callout_templates
```

Comment:
It might be worth caching `app_callout_templates` if it's now going to be called often.